lib/ical/recur_expansion.js

```diff
@@ -198,6 +198,21 @@ class RecurExpansion {
     }
   }
 
+  /**
+   * Compare two ICAL.Time objects.  When the second parameter is a timeless date
+   * and the first parameter is date-with-time, strip the time and compare only
+   * the days.
+   *
+   * @private
+   * @param {ICAL.Time} a   The one object to compare
+   * @param {ICAL.Time} b   The other object to compare
+   */
+  _compare_special(a, b) {
+    if (!a.isDate && b.isDate)
+      return new Time({ year: a.year, month: a.month, day: a.day }).compare(b);
```

Comment:
This also strips out the timezone, which might affect the comparison. Can you add zone handling here?

Reply:
A test or two around this case and zone handling would also be useful.

Reply:
I have added a test.

I am not going to add time zone handling, as  I do not understand how it would work.  This patch fixes a problem I faced in 2022.  I do not work with iCalendar in JavaScript currently.

Reply:
> This also strips out the timezone, which might affect the comparison. Can you add zone handling here?

The commit message contains:

>  When EXDATE is in DATE format and DTSTART is DATE-TIME, then  to determine whether an occurrence shall be excluded, the occurrence shall be converted to DATE and then compared to EXDATE.

I would say in the conversion of VALUE=DATE-TIME to VALUE=DATE the timezone is irrelevant.